docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,448 @@
+
+- *Quality and relevance* - to achieve good retrieval results and avoid postprocessing and reranking, embedding generation techniques and parameters need to be selected carefully
+- *Scalability with high-dimensional data* - rich data increases computational complexity and resource requirements, especially when working with large datasets
+- *Interpretability* - identifying underlying patterns and relationships (e.g., how specific users respond to certain ad types) embedded in abstract vectors can be tricky
+
+## Smarter vectors
+
+We can use Superlinked's framework to overcome these challenges by creating vectors that are smarter representations of your data and therefore let you retrieve high quality, actionable insights (e.g., understanding why different users respond to different ad creatives) without postprocessing or reranking.
+
+Let's walk through how you can perform user acquisition analytics on ad creatives using Superlinked library elements, namely:
+
+- **Recency space** - to understand the freshness of information (e.g., signup dates)
+- **Number space** - to interpret user activity (e.g., API calls/day)
+- **TextSimilarity space** - to interpret the semantic meaning of text data (e.g., ad creatives)
+
+## User data
+
+We have data from two 2023 ad campaigns - one from August (with more generic ad messages), and another from December (assisted by a made-up influencer, "XYZCr$$d"). Our data (for 8000 users) includes:
+
+1. signup date, as unix timestamp
+2. the ad creative a user clicked on before signing up
+3. average daily activity (for users who signed up by clicking on a campaign ad_creative), measured in API calls/day (over the user's lifetime)
+
+To make our ad campaigns smarter, **we want to know which users to target with which kinds of ad messaging**. We can discover this by embedding our data into a vectorspace, where we can cluster users and find meaningful groups - using a UMAP visualization to examine the cluster labels' relationship to features of the ad creatives.
+
+Let's get started.
+
+## Setup
+
+Before you do your installations and imports, make sure you have access to Llama2 in your environment. If you're working in google colab, sign in / up on [huggingface](https://huggingface.co/), get an access token [here](https://huggingface.co/settings/tokens/), and login:
+
+```python
+from huggingface_hub import notebook_login
+notebook_login() 
+```
+
+Now, let's get going on our installations and imports.
+
+First, we install superlinked and umap.
+
+```python
+%pip install superlinked==9.32.1
+%pip install umap-learn
+```
+
+Next, import all our components and constants.
+
+(Note: Omit `alt.renderers.enable(“mimetype”)` if you’re running this in [google colab](https://colab.research.google.com/github/superlinked/superlinked/blob/main/notebook/recommendations_e_commerce.ipynb). Keep it if you’re executing in [github](https://github.com/superlinked/VectorHub/blob/main/docs/articles/ecomm-recys.md).)
```

Comment:
this is now solved, the notebook code will infer the environment and use the right renderer. This can be removed